experimental/generation/generator/packages/library/templates/generator.lg

```diff
@@ -342,6 +334,79 @@ ${setOrPushOperation()},
     "value": "=@${entity}"          
 }```
 
+# isWithMinMax()
+- IF: ${propertySchema.minimum || propertySchema.maximum}
+    -```
+{
+    "$kind": "Microsoft.IfCondition",
+    "condition": "@${entity}.number < dialogClass.schema.properties['${property}'].minimum",
+    "actions": [
+        {
+            "$kind": "Microsoft.SendActivity",
+            "activity": "\${${property}_IsTooSmall(@${entity}.number)}"
+        }
+    ],
+    "elseActions": [
+        {
+            "$kind": "Microsoft.IfCondition",
+            "condition": "@${entity}.number > dialogClass.schema.properties['${property}'].maximum",
+            "actions": [
+                {
+                    "$kind": "Microsoft.SendActivity",
+                    "activity": "\${${property}_IsTooBig(@${entity}.number)}"
+                }
+            ],
+            "elseActions": [
+                {
+                    "$kind": "Microsoft.SendActivity",
+                    "activity": "\${confirmationText('${property}',  @${entity}.number)}"
+                },
+            ${setOrPush()}
+            ]
+        }
+    ]
+}``` 
+- ELSEIF: ${propertySchema.items.minimum || propertySchema.items.maximum}
```

Comment:
> ${propertySchema.items.minimum || propertySchema.items.maximum} [](start = 9, length = 64)

You can make this a simple ELSE with just SetOrPush in it.  Items properties are promoted to the parent schema--that is a relatively recent change which is why the old code likely looked at both.  The reason this was done is because we wanted to allow a ref inside items and that includes $entities, $templates, etc.

Reply:
Does it mean that we could remove the condition ${propertySchema.items.minimum || propertySchema.items.maximum} ?

Reply:
Yes.  If they are present they are promoted to the property level.

---
In reply to: [565887606](https://github.com/microsoft/BotBuilder-Samples/pull/3042#discussion_r565887606) [](ancestors = 565887606)